pkg/kubeapiserver/options/authentication.go

```diff
@@ -406,28 +400,30 @@ func (s *BuiltInAuthenticationOptions) ToAuthenticationConfig() (kubeauthenticat
 	return ret, nil
 }
 
-func (o *BuiltInAuthenticationOptions) ApplyTo(c *genericapiserver.Config) error {
+func (o *BuiltInAuthenticationOptions) ApplyTo(c *genericapiserver.Config) (dynamiccertificates.CAContentProvider, error) {
 	if o == nil {
-		return nil
+		return nil, nil
 	}
 
-	if o.ClientCert != nil {
-		clientCertificateCAContentProvider, err := o.ClientCert.GetClientCAContentProvider()
+	var clientCertificateCAContentProvider dynamiccertificates.CAContentProvider
+	if o.ClientCert != nil && len(o.ClientCert.ClientCA) > 0 {
+		var err error
+		clientCertificateCAContentProvider, err = dynamiccertificates.NewDynamicCAContentFromFile("client-ca-bundle", o.ClientCert.ClientCA)
 		if err != nil {
-			return fmt.Errorf("unable to load client CA file: %v", err)
+			return nil, fmt.Errorf("unable to load client CA file: %v", err)
 		}
 		if err = c.Authentication.ApplyClientCert(clientCertificateCAContentProvider, c.SecureServing); err != nil {
```

Comment:
in comparison to @tnozicka 's approach, this has left an authentication.ApplyTo that doesn't actually create a full authenticator.  This method is setting a few pieces and then more pieces are built above and set separate, with the same ordering problem as the other PR.  This PR is leaving partially built authenticators and not resolving any fundamental ordering problem.